internal/profile/install_profile.go

```diff
@@ -0,0 +1,202 @@
+// Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+// or more contributor license agreements. Licensed under the Elastic License;
+// you may not use this file except in compliance with the Elastic License.
+
+package profile
+
+import (
+	"fmt"
+	"io/ioutil"
+	"os"
+	"path/filepath"
+
+	"github.com/pkg/errors"
+
+	"github.com/elastic/elastic-package/internal/configuration/locations"
+	"github.com/elastic/elastic-package/internal/logger"
+)
+
+// CreateProfile creates an existing profile from the default elastic-package config dir
+// if option.Package has a value it'll be used, if not, the default location will be used
+// if option.From does not have a supplied value, it'll create a default profile.
+func CreateProfile(options Options) error {
+	if options.PackagePath == "" {
+		loc, err := locations.NewLocationManager()
+		if err != nil {
+			return errors.Wrap(err, "error finding stack dir location")
+		}
+		options.PackagePath = loc.StackDir()
+	}
+
+	if options.FromProfile == "" {
+		return createProfile(options)
+	}
+
+	return createProfileFrom(options)
+}
+
+// LoadProfile loads an existing profile from the default elastic-package config dir
+func LoadProfile(profileName string) (*Profile, error) {
+	loc, err := locations.NewLocationManager()
+	if err != nil {
+		return nil, errors.Wrap(err, "error finding stack dir location")
+	}
+
+	return loadProfile(loc.StackDir(), profileName)
+}
+
+// DeleteProfile deletes a profile from the default elastic-package config dir
+func DeleteProfile(profileName string) error {
+	loc, err := locations.NewLocationManager()
+	if err != nil {
+		return errors.Wrap(err, "error finding stack dir location")
+	}
+	return deleteProfile(loc.StackDir(), profileName)
+}
+
+// FetchAllProfiles returns a list of profile values
+func FetchAllProfiles(elasticPackagePath string) ([]Metadata, error) {
+	dirList, err := ioutil.ReadDir(elasticPackagePath)
+	if err != nil {
+		return []Metadata{}, errors.Wrapf(err, "error reading from directory %s", elasticPackagePath)
+	}
+
+	var profiles []Metadata
+	// TODO: this should read a profile.json file or something like that
+	for _, item := range dirList {
+		if !item.IsDir() {
+			continue
+		}
+		profile, err := loadProfile(elasticPackagePath, item.Name())
+		if errors.Is(err, ErrNotAProfile) {
+			continue
+		}
+		if err != nil {
+			return profiles, errors.Wrapf(err, "error loading profile %s", item.Name())
+		}
+		metadata, err := profile.metadata()
+		if err != nil {
+			return profiles, errors.Wrap(err, "error reading profile metadata")
+		}
+		profiles = append(profiles, metadata)
+	}
+	return profiles, nil
+}
+
+// createProfile installs a new profile at the given package path location.
+// overwriteExisting determines the behavior if a profile with the given name already exists.
+// On true, it'll overwrite the profile, on false, it'll copy the existing profile over to profileName_old
```

Comment:
this comment is a bit misleading. Maybe it could be simplified that elastic-package will backup the profile if it exists.